Sources/CryptoSwift/HashProtocol.swift

```diff
@@ -16,6 +16,10 @@ internal protocol HashProtocol: class {
 extension HashProtocol {
     
     func prepare(_ len:Int) -> Array<UInt8> {
+        return self.prepare(len, allowance: 8)
+    }
+    
+    func prepare(_ len:Int, allowance: Int) -> Array<UInt8> {
```

Comment:
Agreed. Please keep it then, and remove `prepare(len:)`. This way it will be explicit.

speaking of, it may be good idea to rename this function to `bitPadding` (as this is what it does), and move it  to Utils.swift